packages/cli-hooks/src/protocols.js

```diff
@@ -0,0 +1,119 @@
+import minimist from 'minimist';
+
+/**
+ * An interface encapsulating a specific set of communication rules that the
+ * SDK and the CLI both implement.
+ * @typedef {object} Protocol
+ * @property {string} name -
+ *   Label representing the name of the protocol as expected by the Slack CLI.
+ * @property {typeof console.log} log -
```

Comment:
Wow does using `typeof` actually work for setting a type in JSDoc?

Reply:
Yes! JSDoc has surprised me in such a great way! 🤩

Though I'm now not so sure the `.log` part is strictly honored. It seems like any of these console loggers meets the logging interface so passes type checks (which is solid enough IMO). This is what the `log` type is showing for me 😄 

```
(property) log: {
    (...data: any[]): void;
    (message?: any, ...optionalParams: any[]): void;
    (message?: any, ...optionalParams: any[]): void;
}
─────────────────────────────────────────────────────────────────────────
Logging utility to surface diagnostic info from the SDK or username code.
```

Reply:
That looks correct, those are overloads for the `log` method in particular.